src/data/file_preprocessing/echo_files_parser.py

```diff
@@ -0,0 +1,115 @@
+from __future__ import annotations
+import pandas as pd
+
+class EchoFilesParser:
+    def __init__(
+        self,
+        echo_files: list[str],
+    ):
+        """
+        :param echo_files: list of echo file names
+        """
+        self.echo_files = echo_files
+        self.echo_df = pd.DataFrame()
+        self.exceptions_df = pd.DataFrame()
+
+    def find_marker_rows(self, file:str, markers:tuple[str]) -> list[int]:
+        """
+        Finds the row numbers of marker lines in a file.
+
+        :param file: file to search
+        :param markers: markers to search for
+        """
+        with open(file) as f:
+            markers_rows = list()
+            for i, line in enumerate(f):
+                if line.strip() in markers:
+                    markers_rows.append(i)
+                if len(markers_rows) == len(markers):
+                    return markers_rows
+        if len(markers_rows) == 0:
+            raise ValueError('No marker found in file.')
+        return markers_rows
+
+    def parse_files(self) -> EchoFilesParser:
+        """
+        Preprocesses csv echo files, splits regular records from exceptions.
+        """
+        if not(all(file.endswith('.csv') for file in self.echo_files)):
+            raise ValueError(f"Expected files to be of '*.csv' type, provided: {self.echo_files}")
+
+        for filename in self.echo_files: 
+            markers = self.find_marker_rows(filename, ('[EXCEPTIONS]', '[DETAILS]'))
+
+            if len(markers) == 2:
+                exceptions_line, details_line = markers
+                exceptions_df = pd.read_csv(filename, skiprows=exceptions_line+1, nrows=(details_line-1)-exceptions_line-2)
+                echo_df = pd.read_csv(filename, skiprows=details_line+1)
+            else:
+                exceptions_df = pd.DataFrame()
+                echo_df = pd.read_csv(filename, skiprows=markers[0]+1)
+
+            echo_df = echo_df[~echo_df[echo_df.columns[0]].str.lower().str.startswith('instrument')]
+
+            self.exceptions_df = exceptions_df if self.exceptions_df.empty else pd.concat([self.exceptions_df, exceptions_df], ignore_index=True)
```

Comment:
performance tip: from my experiance it will be way faster (especially for larger number of files) if you aggregate the dataframes in a list and concat them in the end instead of reconcating them on every loop iteration, i.e.

```py
exception_dfs = []
for filename in self.echo_files:
    ...
    exception_dfs.append(exception_df)
merged_exception_dfs = pd.concat(exception_dfs, ignore_index=True)
```

and the same for echo dfs